### DISCUSSION TITLE
How to flat "node_modules" for specific folder in PNPM workspace environment

### DISCUSSION THREAD

**Discussion by tal-rofe** [pnpm/pnpm#7757] [Q&A] [answered]
I am using PNPM workspace, with `pnpm-workspace.yaml` file:
```yaml
packages:
    - 'apps/*'
    - 'functions/*'
```

Now, I have also this `.npmrc` file:
```
save-exact="true"
stream="true"
engine-strict="true"
ignore-workspace-root-check="true"
```

I do use the workspace as required and everything works great.
But, there is some use case when I need to flatten one of the workspace folder's `node_modules` directory.

As PNPM functionates, each workspace folder would have `node_modules` with symlinks. I don't want it - I simply want specific `node_modules` to have the exact packages the folder needs.

To be more specific, this is my project, open-sourced: https://github.com/tal-rofe/url-shortener

If you'd run `pnpm i` on the root (as needed), you can see that the folder `functions/store-url` has its `node_modules` but contains symlinks within.

There is an external tool, AWS Lambda, when I need this complete `node_modules` (no symlinks at all, as I would create this project with the simplest configuration with NPM only for example). As if `functions/store-url` was single folder with its own `package.json` and `package-lock.json`. How can I do it?

**Comment by zkochan** [comment, marked as answer]
Exclude that project from the workspace and run install separately for it with the configuration you need.
